File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Api/Contracts/ReservationDetailsDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Api.Contracts;

/// <summary>
///     Vehicle and reservation details.
/// </summary>
public sealed record ReservationDetailsDto(
    Guid VehicleId,
    string CategoryCode,
    DateOnly PickupDate,
    DateOnly ReturnDate,
    string PickupLocationCode,
    string DropoffLocationCode);

File: src/frontend/libs/reservation-api/src/lib/models/reservation-details.dto.ts
/**
 * Vehicle and reservation details
 */
import type { ISODateString } from "@orange-car-rental/shared";
import type { VehicleId, CategoryCode } from "@orange-car-rental/vehicle-api";
import type { LocationCode } from "@orange-car-rental/location-api";

export type ReservationDetails = {
  readonly vehicleId: VehicleId;
  readonly categoryCode: CategoryCode;
  readonly pickupDate: ISODateString;
  readonly returnDate: ISODateString;
  readonly pickupLocationCode: LocationCode;
  readonly dropoffLocationCode: LocationCode;
};
